src/api-sdk/github/github.ts

```diff
@@ -78,6 +80,29 @@ export class Github {
         return this.responseHandler<any>(response);
     }
 
+    public async postPullRequest (repoUrl: string, body: GitHubPullRequest): Promise<ApiResponse<any> | ApiErrorResponse> {
+        const shaResponse = await this.getData(getShaParams(repoUrl));
+        const baseSha = extractBaseShaHelper(shaResponse);
+
+        const { branchUrl, branchBody } = createBranchParams(repoUrl, body.branchName, baseSha);
+        await this.postData(branchUrl, branchBody);
+
+        const { blobUrl, blobBody } = createBlobParams(repoUrl, body.prContent);
+        const blobSha = extractShaHelper(await this.postData(blobUrl, blobBody));
+
+        const { treeUrl, treeBody } = createTreeParams(repoUrl, baseSha, body.filePath, blobSha);
+        const treeSha = extractShaHelper(await this.postData(treeUrl, treeBody));
+
+        const { commitUrl, commitBody } = createCommitParams(repoUrl, body.prTitle, treeSha, baseSha);
+        const commitSha = extractShaHelper(await this.postData(commitUrl, commitBody));
+
+        const { refUrl, refBody } = updateBranchReferenceParams(repoUrl, body.branchName, commitSha);
+        await this.postData(refUrl, refBody);
+
+        const { prUrl, prPostbody } = createPullRequestParams(repoUrl, body.prTitle, body.prBody, body.branchName, 'main');
```

Comment:
Do not use hardcoded value, please use the constant defaultBranch, change the  `createPullRequestParams` and update the type for `GitHubPullRequest`

```
export const createPullRequestParams = (repoUrl: string, prTitle: string, prBody: string, branchName: string, baseBranch: string = defaultBranch) => {
    const prUrl = `${repoUrl}/pulls`;
    const prPostbody = {
        title: prTitle,
        body: prBody,
        head: branchName,
        base: baseBranch
    };
    return { prUrl, prPostbody };
};
```